### PR TITLE
Add dockerfile for OC-Mirror

### PR DESCRIPTION
**Pull request by janboll**
This creates a Dockerfile containing oc-mirror end required dependencies.

Additionally it adds some documentation on how to use/sync images using oc-mirror

Solves: https://issues.redhat.com/browse/ARO-8968

### CODE REVIEW

**Comment by pontg** [comment]
lgtm

File: image-sync/oc-mirror/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    -o oc.tar.gz && \
+    tar -zvxf oc.tar.gz && \
+    mv oc kubectl /usr/local/bin
+
+RUN curl -sfL https://mirror.openshift.com/pub/openshift-v4/clients/ocp/${OC_VERSION}/oc-mirror.tar.gz \
+    -o oc-mirror.tar.gz && \
+    tar -zvxf oc-mirror.tar.gz && \
+    mv oc-mirror /usr/local/bin
+
+FROM registry.access.redhat.com/ubi9/ubi:9.4-1123.1719560047
```
Comment (geoberle) [inline image-sync/oc-mirror/Dockerfile:17]: the frontend dockerfile in this repo uses MSFT mariner base images. should we use them as well?

https://github.com/Azure/ARO-HCP/blob/main/frontend/Dockerfile

Comment (janboll) [reply image-sync/oc-mirror/Dockerfile:17]: I thought to change this to our mirrored image later. So, mirror this ubi image and build this container based on the ACR image.

Comment (janboll) [reply image-sync/oc-mirror/Dockerfile:17]: Going to switch it to: mcr.microsoft.com/cbl-mariner/distroless/base:2.0-nonroot@sha256:ef0dc582fc2a8dd34fbb41341a3a9a1aaa70d4542ff04ce4e33a641e52e4807e

**Comment by bennerv** [comment]
Would you mind adding a Makefile target for the new component? 

What's the reasoning we would build an image vs. just pull the sources on run in a containerfile?

**Comment by janboll** [comment]
@bennerv I can add a Make target.

 Two your second question, that's a good practice. If the source is not available, the job fails. Additionally, the runtime is increased. It might be my lack of experience in the ARO world, but I do not see why you would not want to do it.

**Review by geoberle — APPROVED** [review]

File: image-sync/oc-mirror/Makefile
Comment (mjlshen) [inline image-sync/oc-mirror/Makefile:1]: This looks good to me, I think we will want to add CI/CD for these make targets as well like https://github.com/Azure/ARO-HCP/blob/92e2037bc7cfaa65014034f03b761d51473d7753/.github/workflows/aro-hcp-dev-env-cd.yml#L117-L149

**Comment by bennerv** [comment]
@janboll was just wondering!  Makes sense, thanks!
> Two your second question, that's a good practice. If the source is not available, the job fails. Additionally, the runtime is increased. It might be my lack of experience in the ARO world, but I do not see why you would not want to do it.

File: image-sync/oc-mirror/Makefile
Comment (janboll) [reply image-sync/oc-mirror/Makefile:1]: Added CI/CD as suggested

File: .github/workflows/aro-hcp-dev-env-cd.yml [hunk 2]
```diff
@@ -157,6 +158,40 @@
...
+
+        - name: 'Az CLI login'
+          if: github.event.pull_request.merged == true
+          uses: azure/login@v2
+          with:
+            client-id: ${{ secrets.AZURE_CLIENT_ID }}
+            tenant-id: ${{ secrets.AZURE_TENANT_ID }}
+            subscription-id: ${{ secrets.AZURE_SUBSCRIPTION_ID }}
+
+        - name: Push frontend container image
```
Comment (mjlshen) [inline .github/workflows/aro-hcp-dev-env-cd.yml:186]:
```suggestion
        - name: Push image-sync container image
```

**Review by mjlshen — COMMENTED** [review]
last small nit :D lgtm overall

**Review by mjlshen — APPROVED** [review]

File: .github/workflows/aro-hcp-dev-env-cd.yml [hunk 3]
```diff
@@ -157,6 +158,40 @@
...
+          uses: azure/login@v2
+          with:
+            client-id: ${{ secrets.AZURE_CLIENT_ID }}
+            tenant-id: ${{ secrets.AZURE_TENANT_ID }}
+            subscription-id: ${{ secrets.AZURE_SUBSCRIPTION_ID }}
+
+        - name: Push image-sync container image
+          if: github.event.pull_request.merged == true
+          env:
+            ARO_HCP_IMAGE_ACR: devarohcp
```
Comment (geoberle) [inline .github/workflows/aro-hcp-dev-env-cd.yml:189]: The new long-living ACR is named arohcpdev (global RG)

File: image-sync/oc-mirror/Makefile [hunk 4]
```diff
@@ -0,0 +1,14 @@
+SHELL = /bin/bash
+COMMIT = $(shell git rev-parse --short=7 HEAD)$(shell [[ $$(git status --porcelain) = "" ]])
+ARO_HCP_BASE_IMAGE ?= devarohcp.azurecr.io
```
Comment (geoberle) [inline image-sync/oc-mirror/Makefile:3]: Have a look how other makefiles import dev-infrastructure/configurations/dev.mk. There is a var in there for the ACR name

**Review by mjlshen — APPROVED** [review]

**Pull request merged**